client/src/utils/FetchUtil.ts

```diff
@@ -1,7 +1,34 @@
 import { DataFile, DataSource } from '@dekkai/data-source';
 import s3Client from '@/services/S3Service';
 
+const storeSizeLimit = 104857600; // 100MB in bytes
+const storeEntryLimit = 10485760; // 10MB in bytes
 const memoizedStore = new Map();
+const memoizedSize = new Map();
+
+const storeSize = (): number => {
+  let totalValue = 0;
+  for (const value of memoizedSize.values()) {
+    totalValue += value;
+  }
+  return totalValue;
+};
+
+const storeResult = (hash: string, result: unknown, size: number): void => {
+  if (size > storeEntryLimit) {
+    return;
+  }
+
+  // TO-DO: Optimize this
```

Comment:
Consider batching deletes so you don't need to free/defragment memory after each call.

Reply:
short of using `map.clear` there isnt a way of batch deleting with js maps anyway, correct me if im wrong... while loop is probably as good as it will be